src/main/java/com/powsybl/openloadflow/sa/WoodburyDcSecurityAnalysis.java

```diff
@@ -201,8 +201,10 @@ protected SecurityAnalysisResult runSimulations(LfNetwork lfNetwork, List<Propag
             // this is a difference with dc security analysis
             cleanContingencies(lfNetwork, propagatedContingencies);
 
+            // compute the pre-contingency states
             double[] preContingencyStates = DcLoadFlowEngine.run(context, new DisabledNetwork(), reportNode);
-            double[] baseContingencyStates = preContingencyStates.clone();
+            // create workingContingencyStates that will be a working copy of pre-contingency states
+            double[] workingContingencyStates = preContingencyStates.clone();
```

Comment:
Use `System.arraycopy` instead of clone ?

Reply:
Done ! I did not know about this subtle difference but seems slightly better indeed (https://stackoverflow.com/questions/7179251/is-there-any-reason-to-prefer-system-arraycopy-over-clone)

Reply:
That was true for Java5 in 2006 but if you run the bench today you see no difference after the JIT has comleted.